src/SFA.DAS.DownloadService.UnitTests/Web.Controllers/AparControllerTests.cs

```diff
@@ -28,28 +28,100 @@ public void Init()
         {
             _mockClient.Setup(x => x.GetAparSummary()).ReturnsAsync(new List<AparEntry>());
             var result = await _controller.DownloadCsv();
             var redirectResult = result as RedirectToActionResult;
-            Assert.AreEqual("ServiceUnavailable",redirectResult.ActionName);
-            _mockClient.Verify(x=> x.GetAparSummary(),Times.Once);
+            Assert.AreEqual("ServiceUnavailable", redirectResult.ActionName);
+        }
+
+        public async Task When_DownloadCsv_IsCalled_And_RecordsAreReturnedFromGetAparSummary_GetLatestNonOnboardingOrganisationChangeDate_IsCalled()
+        {
+            var dateUpdated = DateTime.Now.AddDays(-1);
+            _mockClient.Setup(x => x.GetAparSummary()).ReturnsAsync(new List<AparEntry> { new AparEntry() });
+            _mockClient.Setup(x => x.GetLatestNonOnboardingOrganisationChangeDate()).ReturnsAsync(dateUpdated);
+            var result = await _controller.DownloadCsv();
+            _mockClient.Verify(x => x.GetLatestNonOnboardingOrganisationChangeDate(), Times.Once);
         }
 
         [Test]
-        public async Task Csv_RecordsReturnedFromApi_ExpectedCSVDownloaded()
+        public async Task When_DownloadCsv_IsCalled_And_RecordsAreReturnedFromGetAparSummary_ExpectedCSVDownloaded()
         {
             var dateUpdated = DateTime.Now.AddDays(-1);
-            _mockClient.Setup(x => x.GetAparSummary()).ReturnsAsync(new List<AparEntry> {new AparEntry()});
+            _mockClient.Setup(x => x.GetAparSummary()).ReturnsAsync(new List<AparEntry> { new AparEntry() });
             _mockClient.Setup(x => x.GetLatestNonOnboardingOrganisationChangeDate()).ReturnsAsync(dateUpdated);
             var result = await _controller.DownloadCsv();
             var fileDownloadResult = result as FileContentResult;
             var expectedFileName = $"apar-{dateUpdated:yyyy-MM-dd-HH-mm-ss}.csv";
-            Assert.AreEqual("text/csv", fileDownloadResult.ContentType);
-            Assert.AreEqual(expectedFileName,fileDownloadResult.FileDownloadName);
-            _mockClient.Verify(x => x.GetAparSummary(), Times.Once);
-            _mockClient.Verify(x => x.GetLatestNonOnboardingOrganisationChangeDate(), Times.Once);
+
+            Assert.Multiple(() =>
+            {
+                Assert.AreEqual("text/csv", fileDownloadResult.ContentType);
+                Assert.AreEqual(expectedFileName, fileDownloadResult.FileDownloadName);
+            });
+        }
+
+        [Test]
+        public void IndexRoapt_ShouldLogWarning()
+        {
+            // Act
+            _controller.IndexRoapt();
+
+            // Assert
+            _mockLogger.Verify(
+                x => x.Log(
+                    LogLevel.Warning,
+                    It.IsAny<EventId>(),
+                    It.Is<object>(v => v.ToString().Contains("Deprecated endpoint 'roatp' called for AparController")),
```

Comment:
I don't think we need to test for the exact error message, but if you want to do that then it's better to put the message in a get-only property so that it can be referenced

Reply:
I thinks it's fine as is